File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs.sort(key = lambda x: (x[1]))
        n = len(pairs)
        res = 1
        l = pairs[0][1]
        for i in range(1,n):
            if (l < pairs[i][0]):
                res += 1
                l = pairs[i][1]
        return res

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.java
class Solution {
    public int findLongestChain(int[][] pairs) {
        Arrays.sort(pairs, (a, b) -> Integer.compare(a[1],b[1]));
        int n = pairs.length;
        int l = 1;
        int end = pairs[0][1];
        for (int i = 1; i < n; i++) {
            if (end < pairs[i][0]) {
                l += 1;
                end = pairs[i][1];
            }
        }
        return l;
    }
}